modules/Q02_openai_question.sh

```diff
@@ -0,0 +1,128 @@
+    export CHATGPT_RESULT_CNT=0
+
+    # we wait until there arer entries in the question csv
+    while ! [[ -f "${LOG_DIR}"/"${MAIN_LOG_FILE}" ]]; do
+      sleep 10
+    done
+
+    if [[ -f "${LOG_DIR}"/"${MAIN_LOG_FILE}" ]]; then
+      while ! [[ -f  "${CSV_DIR}/q02_openai_question.csv.tmp" ]]; do
+        sleep 3
+      done
+    fi
+    while ! grep -q "Testing phase ended" "${LOG_DIR}"/"${MAIN_LOG_FILE}"; do
+      if [[ "${CHATGPT_RESULT_CNT}" -ge 0 ]]; then
+        ask_chatgpt
+      fi
+      sleep 20
+    done
+    unset OPENAI_API_KEY
+    module_end_log "${FUNCNAME[0]}" "${CHATGPT_RESULT_CNT}"
+  fi
+}
+
+# looks through the modules and finds chatgpt questions inside the csv
+ask_chatgpt() {
+  local GPT_FILE_DIR_="${LOG_PATH_MODULE}""/gpt_files"
+  local GPT_PRIO_=3
+  # default vars
+  local GPT_QUESTION_="" 
+  local CHATGPT_CODE_=""
+  local GPT_RESPONSE_=""
+  local GPT_RESPONSE_CLEANED_=""
+  local GPT_TOKENS_=0
+  local HTTP_CODE_=200
+  local ORIGIN_MODULE_=""
+  print_output "[*] Checking scripts with ChatGPT that have priority ${MINIMUM_GPT_PRIO} or lower" "no_log"
+  mkdir "${GPT_FILE_DIR_}"
+  while IFS=";" read -r COL1_ COL2_ COL3_ COL4_ COL5_ COL6_ COL7_; do
+    SCRIPT_PATH_TMP_="${COL1_}"
+    GPT_ANCHOR_="${COL2_}"
+    GPT_PRIO_="${COL3_//GPT-Prio-/}"
+    GPT_QUESTION_="${COL4_}"
+    GPT_OUTPUT_FILE_="${COL5_}"
+    GPT_TOKENS_="${COL6_//cost\=/}"
+    GPT_RESPONSE_="${COL7_}"
+    GPT_INPUT_FILE_="$(basename "${SCRIPT_PATH_TMP_}")"
+    
+    print_output "[*] Trying to check inside ${ORANGE}${LOG_DIR}/firmware${NC}" "no_log"
+    SCRIPT_PATH_TMP_="$(find "${LOG_DIR}/firmware" -wholename "*${SCRIPT_PATH_TMP_}")"
+    cp "${SCRIPT_PATH_TMP_}" "${GPT_FILE_DIR_}/${GPT_INPUT_FILE_}.log"
+    print_output "[*] Trying to check ${ORANGE} ${SCRIPT_PATH_TMP_} ${NC}with Question ${ORANGE}${GPT_QUESTION_}${NC}" "no_log"
+    print_output "[*]Prio is ${GPT_PRIO_}"  "no_log"
+
+    if [[ -z ${GPT_RESPONSE_}  ]] && [[ ${GPT_PRIO_} -le ${MINIMUM_GPT_PRIO} ]]; then
+      if [ -f "${SCRIPT_PATH_TMP_}" ]; then
+        # add navbar-item for file
+        sub_module_title "${GPT_INPUT_FILE_}"
+        print_output "[*] Asking ChatGPT about ${ORANGE}$(print_path "${SCRIPT_PATH_TMP_}")${NC}" "" "${GPT_FILE_DIR_}/${GPT_INPUT_FILE_}.log"
+        head -n -2 "${CONFIG_DIR}/gpt_template.json" > "${TMP_DIR}/chat.json"
+        CHATGPT_CODE_=$(sed 's/\\//g;s/"/\\\"/g' "${SCRIPT_PATH_TMP_}" | tr -d '[:space:]')
+        printf '"%s %s"\n}]}' "${GPT_QUESTION_}" "${CHATGPT_CODE_}" >> "${TMP_DIR}/chat.json"
+        print_output "[*] The Combined Cost of the OpenAI request / the length is: ${ORANGE}${#GPT_QUESTION_} + ${#CHATGPT_CODE_}${NC}" "no_log"
+        HTTP_CODE_=$(curl https://api.openai.com/v1/chat/completions -H "Content-Type: application/json" \
+          -H "Authorization: Bearer ${OPENAI_API_KEY}" \
+          -d @"${TMP_DIR}/chat.json" -o "${TMP_DIR}/${GPT_INPUT_FILE_}_response.json" --write-out "%{http_code}")
```

Comment:
we need to ensure the return value does not break the strict mode with `disable_strict_mode` before and `enable_strict_mode` afterwards or with `|| true`